api/provider/aws/deploy_read.go

```diff
@@ -0,0 +1,86 @@
+package aws
+
+import (
+	"encoding/json"
+	"fmt"
+
+	"github.com/aws/aws-sdk-go/service/ecs"
+	"github.com/quintilesims/layer0/common/errors"
+	"github.com/quintilesims/layer0/common/models"
+)
+
+func (d *DeployProvider) Read(deployID string) (*models.Deploy, error) {
+	deployModel, err := d.newDeployModel(deployID)
```

Comment:
This seems a little "cart before the horse". The other read implementations follow the pattern:

* (optional) lookup the entities arn
* Describe the aws entity by it's arn or id
* Using the description, pass arguments along to `makeEntityModel`

So in this case, it would be something like:
```
deployFile, err := json.Marshal(...)
...

return makeDeployModel(deployID, deployFile)
```